README.rst

```diff
@@ -818,7 +730,76 @@ host machine.
 
 .. code:: powershell
 
-   sam local invoke --docker-volume-basedir /c/Users/shlee322/projects/test "Ratings"
+   $ sam local invoke --docker-volume-basedir /c/Users/shlee322/projects/test "Ratings"
+
+Advanced installations
+----------------------
+
+Build From Source
+~~~~~~~~~~~~~~~~~
+
+First, install Python(2.7) on your machine, then run the following:
+
+.. code:: bash
+
+   # Clone the repository
+   $ git clone git@github.com/awslabs/aws-sam-cli.git
+
+   # cd into the git
+   $ cd aws-sam-cli
+
+   # pip install the repository
+   $ pip install --user -e .
+
+Install with PyEnv
+~~~~~~~~~~~~~~~~~~
+.. code:: bash
+
+    # Install PyEnv (https://github.com/pyenv/pyenv#installation)
+    $ brew update
+    $ brew install pyenv
+
+    # Initialize pyenv using bash_profile
+    $ echo -e 'if command -v pyenv 1>/dev/null 2>&1; then\n  eval "$(pyenv init -)"\nfi\nexport PATH="~/.pyenv/bin:$PATH"' >> ~/.bash_profile
+    # or using zshrc
+    $ echo -e 'if command -v pyenv 1>/dev/null 2>&1; then\n  eval "$(pyenv init -)"\nfi\nexport PATH="~/.pyenv/bin:$PATH"' >> ~/.zshrc
+
+    # restart the shell
+    $ exec "$SHELL"
+
+    # Install Python 3.6.4
+    $ pyenv install 3.6.4
+    $ pyenv local 3.6.4
+
+    # Install the CLI
+    $ pip install --user aws-sam-cli
+
+    # Verify your installation worked
+    $ sam –version
+
+Troubleshooting
+~~~~~~~~~~~~~~~
+
+Mac Issues
+^^^^^^^^^^
+
+1. **TLSV1_ALERT_PROTOCOL_VERSION**:
+
+If you get an error something similar to:
+
+::
+
+   Could not fetch URL https://pypi.python.org/simple/click/: There was a problem confirming the ssl certificate: [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:590) - skipping
+
+then you are probably using the default version of Python that came with
+your Mac. This is outdated. So make sure you install Python again using
+homebrew and try again:
+
+.. code:: bash
+
+   $ brew install python@2
```

Comment:
Can we just push to use `brew install python` instead? Given the 2.7 deprecation path (yes still a year and a half away but...), it might be best to just push installation to Py3? At very least, we should call out just doing `brew install python` (which is what we already have: https://github.com/awslabs/aws-sam-cli/blob/develop/README.rst#mac-issues)